testing-modules/testing-libraries-2/src/test/java/com/baeldung/greyboxtesting/SalaryCommissionPercentageCalculatorUnitTest.java

```diff
@@ -0,0 +1,41 @@
+package com.baeldung.greyboxtesting;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.math.BigDecimal;
+import java.util.stream.Stream;
+
+import static com.baeldung.greyboxtesting.SalaryCommissionPercentageCalculator.*;
+import static com.baeldung.greyboxtesting.SalaryCommissionPercentageCalculator.Level.*;
+import static com.baeldung.greyboxtesting.SalaryCommissionPercentageCalculator.SalesImpact.*;
+import static com.baeldung.greyboxtesting.SalaryCommissionPercentageCalculator.Seniority.*;
+import static com.baeldung.greyboxtesting.SalaryCommissionPercentageCalculator.Type.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class SalaryCommissionPercentageCalculatorUnitTest {
+
+    private SalaryCommissionPercentageCalculator testTarget = new SalaryCommissionPercentageCalculator();
+
+    @ParameterizedTest
+    @MethodSource("provideSuitableOrthogonalArray")
+    void givenSuitableOrthogonalArray_whenCalculateCommission_thenReturnExpectedResult(Level level, Type type, Seniority seniority, SalesImpact impact, double expected) {
+        BigDecimal got = testTarget.calculate(level, type, seniority, impact);
+        assertEquals(BigDecimal.valueOf(expected), got);
+    }
+
+    private static Stream<Arguments> provideSuitableOrthogonalArray() {
```

Comment:
While `orthogonal` is a reasonable word here, it's a little hard for the average person to understand.